### PR TITLE
Wk2 Feature 2: Intro to Json

### PR DESCRIPTION
**Pull request by mariah-crawford**
Prompt: "Modify DataServlet.java to return some hard-coded JSON data, and then modify script.js to fetch that JSON and build a UI from it"

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,11 +27,49 @@
...
+    json += "\"afternoon\": ";
+    json += "\"" +  messages.get(1) + "\"";
+    json += ", ";
+    json += "\"evening\": ";
+    json += "\"" + messages.get(2) + "\"";
+    json += ", ";
+    json += "\"night\": ";
+    json += "\"" + messages.get(3) + "\"";
+    json += "}";
+    return json;
```
Comment (KidEinstein) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:73]: Let's remove this function now that it is no longer used.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,11 +27,49 @@
...
+    String json = gson.toJson(messages);  
+    
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+
+    // response.setContentType("text/html;");
     //why doesnt the headers in html format work and only show up as chars?
-    response.getWriter().println("Hello Mariah! How are you today?");
+    //response.getWriter().println("Hello Mariah! How are you today?");
```
Comment (KidEinstein) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:53]: Nit: Remove commented out code.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -50,6 +50,42 @@ function addQuoteToDom(quote) {
...
+        /** 
+        msgListElement.appendChild(
+            createListElement('Morning:' + messages[0]));
+        msgListElement.appendChild(
+            createListElement('Afternoon: ' + messages[1]));
+        msgListElement.appendChild(
+            createListElement('Evening: ' + messages[2]));
+        msgListElement.appendChild(
+            createListElement('Night: ' + messages[3]));
+        */
```
Comment (KidEinstein) [inline portfolio/src/main/webapp/script.js:78]: Nit: Remove commented out code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -23,11 +27,49 @@
...
+    messages = new ArrayList<String>();
+    messages.add("Good Morning!");
+    messages.add("Good Afternoon!");
+    messages.add("Good Evening!");
+    messages.add("Good Night!");
+  }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
+    // Convert the ArrayList to JSON
```
Comment (KidEinstein) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Great job commenting your code everywhere!

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -50,6 +50,42 @@ function addQuoteToDom(quote) {
...
+    .then(response => response.json()) //like a function that takes in response then does the following..
+    .then((messages) => { // multiple line function that takes in msg and does the following: 
+        console.log(messages);
+
+        // add hard-coded comments to the page
+        const msgListElement = document.getElementById('messages-container');
+        msgListElement.innerHTML = '';
+        for(i = 0; i < messages.length ; i++){
+            msgListElement.appendChild(
+            createListElement(messages[i]));
```
Comment (KidEinstein) [inline portfolio/src/main/webapp/script.js:67]: You can move this to a single line. If you would like to have it in a new line, like you have right now, make sure that createListElement() is indented properly.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (mariah-crawford) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (mariah-crawford) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (mariah-crawford) [reply portfolio/src/main/webapp/script.js]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (mariah-crawford) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: Done

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (mariah-crawford) [reply portfolio/src/main/webapp/script.js]: Done

**Pull request merged**